### PR TITLE
docs(weave): Update platform doc with dedicated cloud info

### PR DESCRIPTION
**Pull request by J2-D2-3PO**
https://wandb.atlassian.net/browse/DOCS-1119

Update https://weave-docs.wandb.ai/guides/platform/ with info about Dedicated Cloud based off of [Weave platform doc update - 12/2024](https://docs.google.com/document/d/181UEwLqTWU5Ivq9IIh0w4PL0fBVQjddCTUI8C1SGhsM/edit?tab=t.0} per request from @abhinavg6

### CODE REVIEW

File: docs/docs/guides/platform/index.md [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
+- **[SCIM API and Python SDK](https://docs.wandb.ai/guides/hosting/iam/automate_iam):** Manage users and teams efficiently with SCIM API and Python SDK.
 
 ## Data Security
 
-In the W&B SaaS Cloud, data of all Weave users is stored in a shared cloud storage and is processed using shared compute services. The shared cloud storage is encrypted using the cloud-native encryption mechanism. When reading or writing data on behalf of a user, a security context comprising of the user's W&B organization, team and project is utilized to ensure data path isolation.
+- **SaaS Cloud:** Data for all Weave users is stored in a shared Clickhouse Cloud cluster, encrypted using cloud-native encryption. Shared compute services process the data, ensuring isolation through a security context comprising your W&B organization, team, and project.
+
+- **Dedicated Cloud:** Data is stored in a unique Clickhouse Cloud cluster in the cloud and region of your choice. A unique compute environment processes the data, with the following additional protections:
+  - **IP allowlisting:** Authorize access to your instance from specific IP addresses.
```
Comment (abhinavg6) [inline docs/docs/guides/platform/index.md:25]:
```suggestion
  - **[IP allowlisting](https://docs.wandb.ai/guides/hosting/data-security/ip-allowlisting):** Authorize access to your instance from specific IP addresses. This is an optional capability.
```

File: docs/docs/guides/platform/index.md [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
+  - **Private connectivity:** Route data securely through the cloud provider's private network.
```
Comment (abhinavg6) [inline docs/docs/guides/platform/index.md:26]:
```suggestion
  - **[Private connectivity](https://docs.wandb.ai/guides/hosting/data-security/private-connectivity):** Route data securely through the cloud provider's private network. This is an optional capability.
```

File: docs/docs/guides/platform/index.md [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
+  - **Data encryption:** Encrypt data at rest using a unique W&B-managed encryption key.
```
Comment (abhinavg6) [inline docs/docs/guides/platform/index.md:27]:
```suggestion
  - **[Data encryption](https://docs.wandb.ai/guides/hosting/data-security/data-encryption):** W&B encrypts data at rest using a unique W&B-managed encryption key.
```

File: docs/docs/guides/platform/index.md [hunk 4]
```diff
@@ -1,31 +1,44 @@
...
+  - **Strong encryption mechanisms:** File-based encryption using Clickhouse Cloud's capabilities.
```
Comment (abhinavg6) [inline docs/docs/guides/platform/index.md:28]:
```suggestion
  - **Clickhouse cluster security:** W&B connects to the unique Clickhouse Cloud cluster for your Dedicated Cloud instance over the cloud provider's private network. W&B also encrypts the cluster using a unique W&B-managed encryption key, while leveraging Clickhouse's file level encryption.
```

File: docs/docs/guides/platform/index.md [hunk 5]
```diff
@@ -1,31 +1,44 @@
...
 
-:::note
-[Secure storage connector](https://docs.wandb.ai/guides/hosting/secure-storage-connector) is not applicable to Weave.
+:::important
+[The W&B Platform secure storage connector](https://docs.wandb.ai/guides/hosting/data-security/secure-storage-connector/?_gl=1*jzcfye*_gcl_au*OTI3ODM1OTcyLjE3MzE0MzU1NjUuMTYxOTQ4Mzk1LjE3MzMyNTYwMTYuMTczMzI1NjAxNQ..*_ga*ODEyMjQ4MjkyLjE3MzE0MzU1NjU.*_ga_JH1SJHJQXJ*MTczNDEyNjAxNy45Ny4xLjE3MzQxMjg5NDEuNjAuMC4w*_ga_GMYDGNGKDT*MTczNDEyNzgzMy44MS4xLjE3MzQxMjg1MTAuMC4wLjA.) is not available in Weave.
```
Comment (abhinavg6) [inline docs/docs/guides/platform/index.md:31]:
```suggestion
[The W&B Platform secure storage connector or BYOB](https://docs.wandb.ai/guides/hosting/data-security/secure-storage-connector) is not available for Weave.
```

**Review by abhinavg6 — CHANGES_REQUESTED** [review]
Thanks James. Requested a few changes here.

**Review by abhinavg6 — APPROVED** [review]
LGTM

**Comment by J2-D2-3PO** [comment]
@scottire this is the PR I was referring to

**Comment by J2-D2-3PO** [comment]
@wandb/weave-team can I get merge approval please?

**Review by scottire — APPROVED** [review]

**Pull request merged**